lib/fieldhand/paginator.rb

```diff
@@ -86,6 +90,19 @@ def parse_response(query = {})
       response_parser
     end
 
+    def retryable_request(query)
+      response = request(query)
+      raise ResponseError, response unless response.is_a?(::Net::HTTPSuccess)
+
+      response
+    rescue ResponseError => e
+      raise e unless self.retries > 0
```

Comment:
Do you need the explicit `self` both here and on the next line?

Reply:
Apparently (tested on Ruby 2.3+) without setting it explicitly `retries` returns nil here. I am not 100% about the root cause yet

Reply:
I believe you need the explicit `self` for `self.retries -= 1` (as `retries -= 1` will try to read a local variable `retries` which does not exist) but you can drop it from the getter, e.g.

```ruby
attr_accessor :retries
private :retries, :retries=

# ...

rescue ResponseError => e
  raise e unless retries > 0
  self.retries -= 1

  # ...
end
```